apps/sms/js/contacts.js

```diff
@@ -206,31 +200,27 @@
           results = contacts;
         }
 
-        callback(results, {
-          terms: terms
-        });
-      };
-
-      request.onerror = function onerror() {
-        // When an error occurs, regardless of completed count,
-        // clear the onsuccess handler from this request and immediately
-        // invoke the callback with a `null` argument.
-        this.onsuccess = this.onerror = null;
-        callback(null);
-      };
+        return results;
+      }).catch(() => {
+        return null;
+      });
     },
-    findByString: function contacts_findBy(filterValue, callback) {
+
+    findByString: function contacts_findBy(filterValue) {
       var unknownCallback = function(contacts) {
         contacts = contacts || [];
-        this.findByUnknown(filterValue, function(unknown) {
-          unknown = unknown || [];
-          callback(contacts.concat(unknown));
-        });
+        return this.findByUnknown(filterValue)
```

Comment:
Yep, your version is better. ```([known, unknown])``` - didn't know that it's possible, with this ```Promise.all``` callbacks will be so sweet! And ```(known || [])``` is more correct since we still can return ```null``` :)